plugins/item_tasks/server/rest.py

```diff
@@ -168,6 +168,29 @@ def _transformOutputs(self, outputs, token, job):
 
         return transformed
 
+    def _validateOutputParentType(self, outputId, parentType, outputSpec):
+        """
+        Checks if the output parent type is compatible with the output type.
+        """
+
+        # Find the corresponding output specification for the given outputID
+        output = None
+        for out in outputSpec:
+            if outputId == out['id']:
+                output = out
+
+        # If a corresponding output is found, check if its parent type is valid
+        if output:
+            if output['type'] == 'new-file' and parentType not in {'item', 'folder'}:
+                return False
+            elif output['type'] == 'new-folder' and\
+                    parentType not in {'folder', 'user', 'collection'}:
+                return False
+            else:
+                return True
+        else:
+            return False
```

Comment:
This should probably raise its own `ValidationException` since it's not really an invalid parent type, but actually an invalid output ID.